sandy/core/cov.py

```diff
@@ -531,24 +575,33 @@ def decompose(self):
         Example
         -------
         >>> sandy.CategoryCov([[1, 0.4],[0.4, 1]]).decompose()
+        array([[-1.        ,  0.        ],
+               [-0.4       ,  0.91651514]])
+
+        >>> sandy.CategoryCov([[1, 0.4],[0.4, 1]]).decompose(sparse=True)
         array([[-1.        ,  0.        ],
                [-0.4       ,  0.91651514]])
         """
-        E, V = self.eig(sort=False)
+        E, V = self.eig(sort=False, sparse=sparse)
         E[E <= 0] = 0
-        M = V.values.dot(np.diag(np.sqrt(E.values)))
+        if sparse==True:
+            M = sps.csr_matrix(V).dot(sps.diags(np.sqrt(E))).toarray()
+        else:
+            M = V.values.dot(np.diag(np.sqrt(E.values)))
         Q, R = scipy.linalg.qr(M.T)
         return R.T
 
     @classmethod
-    def from_var(cls, var):
+    def from_var(cls, var, sparse=False):

```

Comment:
Why do we need sparse here?

Reply:
I tried to optimize with sparse matrix all the mathematical operations that we have. Creating a diagonal matrix I thought it consume less memory in that way.